docs/enhancements/ipxe-host-boot.md

```diff
@@ -0,0 +1,125 @@
+---
+title: ipxe-host-boot
+authors:
+  - "@carbonin"
+  - “@mhrivnak”
+creation-date: 2022-02-22
+last-updated: 2022-02-22
+---
+
+# iPXE Host Boot
+
+## Summary
+
+Not all user infrastructure is able to boot isos directly, and iPXE is a commonly 
+used alternative for booting machines. We should support this by making the 
+application directly serve iPXE artifacts so that systems can boot with no 
+other iPXE infrastructure required.
+
+## Motivation
+
+Booting a host using iPXE should be a first-class operation when using
+assisted-service. This has been found to be at least as common a method 
+as providing the live ISO via virtual media and is also required in some 
+environments. Users are already unpacking the ISO we provide into PXE 
+artifacts and serving those in their infrastructure. We can make this much 
+easier for them by serving those artifacts as well as the iPXE boot script 
+directly from our application.
+
+### Goals
+
+- Serve iPXE artifacts (kernel, rootfs, and initrd)
+- Serve an iPXE boot script
+- Include the ignition in the provided initrd
+
+### Non-Goals
+
+- Support PXE (tftp based)
+- Integration with ironic (separate effort)
+- iPXE boot script customization (maybe in a follow-up if it’s useful)
+- Configure the minimal ISO to pull the rootfs from the image service
+
+## Proposal
+
+The image-service will serve the kernel, rootfs, and initrd with ignition 
+appended (see Implementation Details), while the assisted service will 
+provide an API to download the boot script. The boot script created and served 
+by the assisted service will contain URLs to the required artifacts hosted by 
+the image service.
+
+### User Stories
+
+#### Story 1
+
+As a user of assisted installer I want to boot my hosts using iPXE without 
+setting up my own http server to host artifacts.
+
+### Implementation Details/Notes/Constraints
+
+#### Ironic Integration
+
+To support integration with Ironic’s existing iPXE boot methods, the 
```

Comment:
will it work for other platforms as well? or something specific to ironic?

Reply:
what other platforms? 
as far as i know this is the only platform that running another agent no? or is there another use case?

Reply:
This just means that we're forced to include the ignition as a part of the initrd as opposed to a separate kernel parameter (as is done in the [fcos docs](https://docs.fedoraproject.org/en-US/fedora-coreos/live-booting/#_booting_via_ipxe), for example).

I don't think it makes much sense for us to do this in the ZTP case and not in others so we'd do it the same no matter what. This is just mentioning how the ironic use-case is forcing the design a particular way.

Reply:
what I meant is if we provide this kind of initrd to equinix server - will it boot? Or it needs ignition seperately

Reply:
Should boot fine.